### PR TITLE
Complete "relay" test

### PR DESCRIPTION
**Pull request by cloudhead**

### CODE REVIEW

File: radicle-node/src/service.rs [hunk 1]
```diff
@@ -771,6 +785,7 @@ where
                 }
 
                 for id in message.inventory.as_slice() {
+                    // FIXME: Move this out (good luck with the borrow checker).
```
Comment (FintanH) [inline radicle-node/src/service.rs:788]: That's what I tell myself every day.

File: radicle-node/src/service.rs [hunk 2]
```diff
@@ -802,38 +809,64 @@ where
...
-                    // the resource. Therefore, it's safe to fetch from the remote
-                    // peer, even though it isn't the announcer.
-                    self.fetch(message.id, relayer);
 
-                    return Ok(true);
+                    // Refs can be relayed by peers who don't have the data in storage,
+                    // therefore we only check whether we are connected to the *announcer*,
+                    // which is required by the protocol to only announce refs it has.
+                    if self.sessions.is_negotiated(announcer) {
+                        let repo = self.storage.repository(message.rid).unwrap();
```
Comment (FintanH) [inline radicle-node/src/service.rs:847]: Intentional `unwrap` here?

File: radicle-node/src/service.rs [hunk 3]
```diff
@@ -1083,39 +1116,48 @@ where
...
+                            "refs announcement limit ({}) exceeded, peers will see only some of your repository references",
+                            REF_REMOTE_LIMIT,
+                        );
+                        break;
+                    }
+                }
+            }
+            Namespaces::One(pk) => refs
+                .push((pk, repo.remote(&pk)?.refs.unverified()))
+                // SAFETY: `REF_LIMIT` is greater than 1, thus the bounded vec can hold at least
```
Comment (FintanH) [inline radicle-node/src/service.rs:1154]: You changed this name, right?
```suggestion
                // SAFETY: `REF_REMOTE_LIMIT` is greater than 1, thus the bounded vec can hold at least
```

File: radicle-node/src/service.rs (hunk 1 above)
Comment (cloudhead) [reply radicle-node/src/service.rs:780]: Hahaha.

File: radicle-node/src/service.rs (hunk 2 above)
Comment (cloudhead) [reply radicle-node/src/service.rs:847]: Hmm nope, good catch!

File: radicle-node/src/service.rs (hunk 3 above)
Comment (cloudhead) [reply radicle-node/src/service.rs:1154]: Indeed, let me fix!

**Pull request merged**